### PR TITLE
File fetcher ticket

### PR DESCRIPTION
**Pull request by nick-mynatt**
A utility to assist the MainLoop with fetching files

### CODE REVIEW

File: ndn_hydra/repo/main/main_loop.py [hunk 1]
```diff
@@ -162,19 +164,7 @@ def store(self, file_name: str):
...
-        self.fetching.append(file_name)
-        self.logger.info(f"[ACT][FETCH]*   fil={file_name};pcks={packets};fetch_path={fetch_path}")
-        start = time.time()
-        async for (_, _, content, data_bytes, key) in concurrent_fetcher(self.app, fetch_path, file_name, 0, packets-1, aio.Semaphore(15)):
-            self.data_storage.put_packet(key, data_bytes) #TODO: check digest
-        end = time.time()
-        duration = end -start
-        self.logger.info(f"[ACT][FETCHED]* pcks={packets};duration={duration}")
-        self.store(file_name)
+        return self.file_fetcher.fetch_file_from_client(file_name, packets, packet_size, fetch_path)
```
[the remark this answers is not in the document]
Comment (nick-mynatt) [reply ndn_hydra/repo/main/main_loop.py]: You're right. Fixed!

File: ndn_hydra/repo/modules/file_fetcher.py [hunk 2]
```diff
@@ -9,5 +9,70 @@
...
+        aio.ensure_future(self._fetch_file_helper(file_name, packets, packet_size, fetch_path))
+
+    async def _fetch_file_helper(self, file_name: str, packets: int, packet_size: int, fetch_path: str):
+        if not self.store_func:
+            self.logger.info("FileFetcher: No storage function defined")
+            return
+        if file_name in self.fetching:
+            self.logger.info("FileFetcher: Already fetching")
+            return
+        self.fetching.append(file_name)
```
[the remark this answers is not in the document]
Comment (nick-mynatt) [reply ndn_hydra/repo/modules/file_fetcher.py]: I agree, that makes much more sense. Fixed in most recent commit.

**Comment by justincpresley** [comment]
[Can you test your pull requests?](https://ndn-hydra.readthedocs.io/src/examples.html)

If you comment the results on the PR you tested, I am can push PRs in faster.

**Comment by nick-mynatt** [comment]
Yes, I have a testing environment set up but I will start uploading the results of the tests.

**Comment by nick-mynatt** [comment]
Tested.
Screenshot running insert, query, fetch
Screenshot from 2022-09-28 17-18-17

**Pull request merged**